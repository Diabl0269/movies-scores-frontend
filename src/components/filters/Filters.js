import React from 'react'
import { Container } from '@material-ui/core'
import QueryTypeSelector from './QueryTypeSelector'
import FieldsSwitch from './form/fields/FieldsSwitch'

export default () => {
  return (
    <Container id="filtersContainer">
      <QueryTypeSelector />
      <FieldsSwitch />
    </Container>
  )
}
