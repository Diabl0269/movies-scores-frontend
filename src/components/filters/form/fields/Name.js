import React from 'react'
import { useFormikContext } from 'formik'
import { TextField } from '@material-ui/core'
import handleFormikChange from '../../../../utils/handleFormikChange'

export default () => {
  const { setFieldValue } = useFormikContext()

  const handleChange = (e) => {
    handleFormikChange(setFieldValue, e)
  }

  return (
    <>
      <TextField name="name" onChange={handleChange} />
    </>
  )
}
