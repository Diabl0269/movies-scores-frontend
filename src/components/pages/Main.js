import React from 'react'
import Filters from '../filters/Filters'
import MoviesRecordsList from '../recordsList/RecordsList'
import Page from '../templates/Page'
import { Container } from '@material-ui/core'
import { MoviesProvider } from '../../context/MoviesProvider'
import d from '../../data/dictionary'

const Content = () => {
  return (
    <Container id="mainPageContentContainer">
      <MoviesProvider>
        <Filters />
        <MoviesRecordsList />
      </MoviesProvider>
    </Container>
  )
}

export default () => {
  return <Page Content={Content} title={d('mainPageTitle')} />
}
