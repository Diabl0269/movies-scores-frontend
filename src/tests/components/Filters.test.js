import React from 'react'
import Filters from '../../components/filters/Filters'
import { mount } from 'enzyme'
import { Select } from '@material-ui/core'
import Name from '../../components/filters/form/fields/Name'
import Main from '../../components/pages/Main'

describe('simulating a query type change', () => {
  const pageWrapper = mount(<Main />)
  const wrapper = pageWrapper.find(Filters)

  it('should match snapshot before change', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should be able to find Name field component', () => {
    const select = wrapper.find(Select)
    select.simulate('change', { target: { value: 'name' } })
    wrapper.update()
    expect(pageWrapper.find(Name).exists).toBeTruthy()
  })
})
