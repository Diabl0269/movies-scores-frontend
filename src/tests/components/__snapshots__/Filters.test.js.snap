// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simulating a query type change should match snapshot before change 1`] = `ReactWrapper {}`;
